content/en/docs/guides/keepalive.md

```diff
@@ -0,0 +1,101 @@
+
+{{% alert title="Note" color="info" %}}
+  There is a related but separate concern called [Health Checking]. Health checking allows server to signal whether a *service* is healthy while keepalive is only about the *connection*.
+{{% /alert %}}
+
+### Background
+
+[TCP keepalive] is a well-known method of maintaining connections and detecting broken connections. When enabled, either side of the connection can send redundant packets, once ACKed by the other side, the connection will be consiered as good. If no ACK after repeated attempts, the connection is deemed broken. Configuration of its variables is provided by the OS on a per-socket level, but is commonly not exposed to higher-level APIs.
+
+TCP keepalive has three parameters to tune:
+
+1. Time (time since last receipt before sending a keepalive).
+2. Interval (interval between keepalives when not receiving reply).
+3. Retry (number of times to retry sending keepalives).
+
+gRPC uses HTTP/2 which provides a mandatory [PING] frame which can be used to estimate round-trip time, bandwidth-delay product, or test the connection. The interval and retry in TPC keepalive don't quite apply to PING because the transport is reliable, so they're replaced with timeout (equivalent to interval * retry) in gRPC PING-based keepalive implementation.
+
+{{% alert title="Note" color="info" %}}
+  It's not required for service owners to support kepalive. **Client authors must coordinate with service owners** for whether a particular client-side setting is acceptable. Service owners decide what they are willing to support, including whether they are willing to receive keepalives at all.
+{{% /alert %}}
+
+### How configuring keepalive affects a call
+
+For unary RPC with quick replies, keepalive is less likely to be triggered. Keepalive would primarily be triggered when there is a long-lived RPC, which will fail if keepalive check failed and the connection was closed.
+
+For steaming RPC, if the connection is closed, any in-progress RPCs will fail. If a call is streaming data, the stream will also be closed and any data that has not yet been sent will be lost.
+
+### How to enable and configure keepalive
+
+Based on language, keepalive can be configured by:
+* `ServerOption` and client `DialOption` in Go.
+* Channel and server builder in Java.
+* Channel and server options in C++ and Pyhton.
+
+{{% alert title="Warning" color="warning" %}}
+  To avoidling DDoS, it's important to take caution while set the keepalive cofigurations. Thus, it is recommended to disables keepalive for HTTP/2 connections with no outstanding streams and for clients to avoid configuring their keepalive much below one minute.
```

Comment:
"To avoidling DDoS" -> "To avoid DDoSing"

"to take caution while set" -> "to take caution when setting"

"to disables keepalive" -> "to disable keepalive"